core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -86,7 +125,17 @@ class SubmissionSupervisor(executionServiceCluster: ExecutionServiceCluster,
   }
 
   private def scheduleNextCheckCurrentWorkflowStatus(actor: ActorRef): Cancellable = {
-    system.scheduler.scheduleOnce(addJitter(submissionPollInterval), actor, CheckCurrentWorkflowStatusCounts)
+    system.scheduler.scheduleOnce(submissionPollInterval, actor, CheckCurrentWorkflowStatusCounts)
+  }
+
+  private def updateGlobalJobExecGauges = {
+    datasource.inTransaction { dataAccess =>
+      dataAccess.submissionQuery.countAllStatuses map { subStatuses =>
+        this.globalSubmissionStatusCounts ++= subStatuses.map{ case (k, v) => SubmissionStatuses.withName(k) -> v }
```

Comment:
This is doing a "get-and-set" from a separate thread, which is not safe even with volatile variables. I would suggest either:
a. Piping this future back to the actor similar to `SaveCurrentWorkflowStatusCounts`, and doing the update from the Actor thread (which is safe).
b. Making the maps ConcurrentMaps instead of volatile: https://www.scala-lang.org/api/current/scala/collection/concurrent/TrieMap.html

Reply:
For a), you're specifically saying do the db call here but then dump the results into a message that we pipe back to self, right?

Reply:
yes